x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/plugin/RestSqlQueryAction.java

```diff
@@ -60,54 +48,9 @@ protected RestChannelConsumer prepareRequest(RestRequest request, NodeClient cli
             sqlRequest = SqlQueryRequest.fromXContent(parser);
         }
 
-        MediaType responseMediaType = sqlMediaTypeParser.getResponseMediaType(request, sqlRequest);
-        if (responseMediaType == null) {
-            String msg = String.format(Locale.ROOT, "Invalid response content type: Accept=[%s], Content-Type=[%s], format=[%s]",
-                request.header("Accept"), request.header("Content-Type"), request.param("format"));
-            throw new IllegalArgumentException(msg);
-        }
-
-        /*
-         * Special handling for the "delimiter" parameter which should only be
-         * checked for being present or not in the case of CSV format. We cannot
-         * override {@link BaseRestHandler#responseParams()} because this
-         * parameter should only be checked for CSV, not always.
-         */
-        if ((responseMediaType instanceof XContentType || ((TextFormat) responseMediaType) != TextFormat.CSV)
-            && request.hasParam(URL_PARAM_DELIMITER)) {
-            throw new IllegalArgumentException(unrecognized(request, Collections.singleton(URL_PARAM_DELIMITER), emptySet(), "parameter"));
-        }
-
-        long startNanos = System.nanoTime();
```

Comment:
I am not sure I agree with this code being "externalized" in a separate class that is a `RestResponseListener` called `SqlResponseFormatter`. If it's a Listener, this should be reflected in its name imo. Also, why its a separate class? Maybe an internal class in RestSqlQueryAction if you really like to refactor the code (to me it looks like a forced refactoring without any real benefit, maybe it's just me)?

Reply:
> [...] class that is a RestResponseListener called SqlResponseFormatter

Good point, I've renamed it to `SqlResponseListener`.

> why its a separate class?

The listener is used by both `RestSqlQueryAction` and `RestSqlAsyncGetResultsAction`.